fabric/bin/peer.sh

```diff
@@ -331,25 +331,46 @@ handle_channel_join() {
     try $RUN ${FABRIC_BIN_DIR}/peer "${ARGS_EXEC[@]}"
 
     # - handle ercc
-    say "Installing & Instantiating ercc on channel '${CHAN_ID}' ..."
-    #   - install ercc
-    try $RUN ${FABRIC_BIN_DIR}/peer chaincode install -n ${ERCC_ID} -v ${ERCC_VERSION} -p github.com/hyperledger-labs/fabric-private-chaincode/ercc/cmd
-    sleep 1
-    #   - instantiate ercc iff "channel creation" peer
-    if [ -e "${FABRIC_STATE_DIR}/${CHAN_ID}.creator" ]; then
-	try $RUN ${FABRIC_BIN_DIR}/peer chaincode instantiate -n ${ERCC_ID} -v ${ERCC_VERSION} -c '{"args":["init"]}' -C ${CHAN_ID} -V ercc-vscc
-	sleep 4
-	try rm "${FABRIC_STATE_DIR}/${CHAN_ID}.creator"
-    fi
-    #   - get SPID (mostly as debug output)
-    try $RUN ${FABRIC_BIN_DIR}/peer chaincode query -n ${ERCC_ID} -c '{"args":["getSPID"]}' -C ${CHAN_ID}
+    say "Installing ercc on channel '${CHAN_ID}' ..."
+    say "Packaging chaincode ..."
+    try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode package ${FABRIC_STATE_DIR}/ercc.tar.gz -p ../../ercc --lang golang --label erccv1 
+    para
+    sleep 3
+    say "In installing chaincode ..."
+    try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode install ${FABRIC_STATE_DIR}/ercc.tar.gz 
+    para
+    sleep 3
+    say "Query installed chaincodes.."
+    try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode queryinstalled >&log.txt
+    para
+    PACKAGE_ID=$(awk '/Package ID:/{print}' log.txt | sed -n 's/^Package ID: //; s/, Label:.*$//;p')
+    say "Approve for my org"
+    try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode approveformyorg -o ${ORDERER_ADDR} --channelID ${CHAN_ID} --name ercc --version 1 --package-id ${PACKAGE_ID} --sequence 1 
+    para
+    sleep 3
+    say "Check for commit readiness"
+    try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode checkcommitreadiness --channelID ${CHAN_ID} --name ercc --version 1 --sequence 1 --output json 
```

Comment:
also noticed only now: the instantiate had previously a '-V ercc-vscc' argument.  Right now we don't have the plugin but we still will need it once it is re-enabled. The argument is still the same format and will have to be passed to all of 'approveformyorg', 'checkcommitreadiness' and 'commit'. 
```suggestion
    try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode checkcommitreadiness --channelID ${CHAN_ID} --name ercc --version 1 --sequence 1 --output json # -V ercc-vscc   # TODO: re-add validation plugin once they are enabled in peer ...
```